File: P1S3/CadenaFiltros.cpp
#include "include/CadenaFiltros.h"

void CadenaFiltros::ejecutar(Notas &notas) {
    for(const std::reference_wrapper<Filtro>& f : cadenaFiltros) {
        f.get().ejecutar(notas);
    }
}


void CadenaFiltros::addFiltro(Filtro& filtro) {
    cadenaFiltros.emplace_back(filtro);
}

File: P1S3/dart/cadena_filtros.dart
import 'filtro.dart';
import 'notas.dart';

class CadenaFiltros {
  final _cadenaFiltros = <Filtro>[];

  void ejecutar(Notas notas) {
    for(var f in _cadenaFiltros) {
      f.ejecutar(notas);
    }
  }

  void addFiltro(Filtro filtro) {
    _cadenaFiltros.add(filtro);
  }
}